CONTRIBUTING.md

```diff
@@ -0,0 +1,379 @@
+# How to contribute to Arcs
+
+Thank you for taking the time to contribute to the Arcs project. 
+This guide outlines how to become an Arcs developer.
+
+**Table of Contents**
+
+[Workflow](#workflow)
+* [Filing Bugs](#filing-bugs)
+* [Contributing a Patch](#contributing-a-patch)
+
+[Developer Installation](#developer-installation)
+* [Additional Bazel Installation](#additional-bazel-installation)
+* [Windows Installation Notes](#windows-installation-notes)
+* [Mac Installation Notes](#mac-installation-notes)
+* [IDE Setup](#ide-setup)
+* [Git Setup](#git-setup)
+
+[Starting Arcs](#starting-arcs)
+
+[Testing](#testing)
+* [WebAssembly tests](#webassembly-tests)
+* [Debugging tests](#debugging-tests)
+  * [Debugging unit tests in Chrome](#debugging-unit-tests-in-chrome)
+* [Debugging WebDriver Failures](#debugging-webdriver-failures)
+  * [Graphical (non-headless)](#graphical-non-headless)
+  * [Attaching a Debugger](#attaching-a-debugger)
+
+
+## Workflow
+
+### Filing Bugs
+
+**If you notice a security issue, please email 
+[arcs-team@google.com](mailto:arcs-team+security@google.com) instead of filing a bug.**
+
+TODO(arcs-team): Devise a way to track bugs internally and in the open.
+
+### Contributing a Patch
+
+1) Create a PR and go through reviews.
+
+   If you are an external contributor, you will have to agree to the 
+   [Contributor License Agreement](https://opensource.google/docs/cla/).
+
+1) When you are ready to merge, add the "ready to pull" label to your PR.
+
+   This will trigger an internal review process for the change. If you are an 
+   internal contributor, please click `+2` to the Safe Review. Otherwise, please
+   reach out to a Googler to help you land the change.
+   
+1) [Copybara](https://github.com/google/copybara) will create another import CL. 
+   Get _another Googler_ to approve the CL.
+   
+1) When GCB and our internal integration tests are green, the import CL will be
+   automatically submitted and the corresponding PR will be merged on GitHub.
+
+## Developer Installation
```

Comment:
What about using the tools/setup script?

Reply:
I'm on the fence about this. I noticed that the instructions for the setup script these days have a lot of additional steps. This differs from the intention of the script, which is to fire once and have the dev environment be ready to go. 

The setup script currently doesn't install Bazel or the Android SDK, it just handles the node runtime. If some cases are manual, why not all? Even for the node case, it doesn't do the best job due to nuances with how nvm is set up. E.g. we require users to re-run the script and/or source profile. 

I'm thinking that we can document the setup script once it truly supports a 1-step setup.

...